server/plugin.go

```diff
@@ -270,6 +274,19 @@ func (p *Plugin) getGitlabUserInfoByMattermostID(userID string) (*gitlab.UserInf
 	}
 
 	userInfo.Token.AccessToken = unencryptedToken
+	newToken, err := p.checkAndRefreshToken(userInfo.Token)
+	if err != nil {
+		return nil, &APIErrorResponse{ID: "", Message: err.Error(), StatusCode: http.StatusInternalServerError}
+	}
+
+	if newToken != nil {
+		userInfo.Token = newToken
+		unencryptedToken = newToken.AccessToken // needed because the storeGitlabUserInfo method changes its value to an encrypted value
```

Comment:
> needed because the storeGitlabUserInfo method changes its value to an encrypted value

I think a safer implementation would be to have `storeGitlabUserInfo` make a copy of the incoming `UserInfo` argument to edit, instead of mutating the argument. This works though :+1: